src/lib/journey/callbacks/password/confirm-input.svelte

```diff
@@ -24,14 +25,26 @@
 
   let isVisible = false;
   let type: 'password' | 'text' = 'password';
-  let value: unknown;
+  let value: Maybe<string>;
+
+  function onChangeWrapper(event: Event) {
+    value = (event.target as HTMLInputElement)?.value;
+    onChange(typeof value === 'string' ? value : undefined);
+  }
   /**
    * @function toggleVisibility - toggles the password from masked to plaintext
    */
   function toggleVisibility() {
     isVisible = !isVisible;
     type = isVisible ? 'text' : 'password';
   }
+
+  $: {
+    if (resetValue) {
```

Comment:
There's a bit of logic in the parent component that helps inform this component of when to clear out its value.